src/sets.jl

```diff
@@ -345,6 +480,15 @@ end
 dual_set(s::NormSpectralCone) = NormNuclearCone(s.row_dim, s.column_dim)
 dual_set_type(::Type{NormSpectralCone}) = NormNuclearCone
 
+function distance_to_set(v::AbstractVector{<:Real}, s::NormSpectralCone)
+    _check_dimension(v, s)
+    t = v[1]
+    m = reshape(v[2:end], (s.row_dim, s.column_dim))
+    s1 = LinearAlgebra.svd(m).S[1]
```

Comment:
I'm not an expert on this, but I know that computing the full SVD is an extremely expensive way to compute the smallest singular value. This is probably not suitable for non-toy applications. There are a few packages that do this, e.g., https://jutho.github.io/KrylovKit.jl/latest/man/svd/. We should evaluate which, if any, are suitable to add as dependencies of MOI.

Reply:
agreed, I just kept the dependency-free version for now. MOI can either add a dependency or vendor (copy and acknowledge) a single function from another package